### ISSUE TITLE
[XLA] using auto-clustering with tf_xla_auto_jit causes multiple compilation with variable seq. len.

### ISSUE THREAD

**Issue by TParcollet** [tensorflow/tensorflow#62198]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.13.1

### Custom code

Yes

### Current behavior?

I have a somewhat complex model that deals with variable length tensors as input. The feature dimension remains the same, only one of the dimension changes (time). If I fix this by giving always the same batch of features, I can use tf_xla_auto_jit and see that the training speed improves quite a lot. However, as soon as I throw the for `batch in train_dataset` in the mix, everything becomes infinitely slow and, looking at the warning, it feels like XLA keeps recompiling ... 

I don't really know what to expect here? 

### Standalone code to reproduce the issue

```shell
Impossible to share unfortunately.
```

**Comment by TParcollet** [comment]
Note: I tried by decorating using `with tf.xla.experimental.jit_scope():`, and the same thing happens, it feels like everything is being re-compiled at every new entry in the scope.

**Comment by maksym33** [comment]
@TParcollet 
From what I remember the XLA compilation does not support ragged tensors (variable length).
I also found that initalizing the ragged tensors can sometimes take quite a lot of time if done through `tf.ragged.constant`. Depends on the use case obviously.

**Comment by TParcollet** [comment]
@maksym33 well, if this is the case how the hell I am supposed to deactivate it... I tried the three versions (argument, context and config) and it STILL triggers XLA. I then compiled from the source without XLA support and then it then crashed with this "Trying to access resource Resource..." error. Any version above 2.12 triggers XLA from one way or another on my code - even when TF is not compiled with it.
